src/sandstorm/grain.capnp

```diff
@@ -467,6 +588,40 @@ interface SessionContext {
   #     // pass event.data.token to your app's server and call SandstormApi.restore() with it
   #   }
   # }, false)
+
+  provide @4 (cap :Capability, requiredPermissions :PermissionSet,
+              descriptor :PowerboxDescriptor, displayInfo :PowerboxDisplayInfo);
+  # For sessions started with `newRequestSession()`, fulfills the original request. If only one
+  # capability was requested, the powerbox will close upon `provide()` being called. If multiple
+  # capabilities were requested, then the powerbox remains open and `provide()` may be called
```

Comment:
> What would it look like for multiple capabilities to be requested?

This is something I'm still thinking about. Basically I think `request()` should probably have an additional boolean parameter called `multiple`. When set, then the request is for a list of capabilities rather than a single one. In this case the powerbox presents the user with a UI for building the list by visiting one or more grains and adding items one by one.

The `requestInfo` would apply to all elements in the list equally. E.g. for requesting a list of contacts, `requestInfo` describes the different contact formats that are accepted. All the items in the list might be of just one of those formats or could be a variety.